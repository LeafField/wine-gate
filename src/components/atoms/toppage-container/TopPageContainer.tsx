import React, { FC, PropsWithChildren } from "react";

type Props = {
  padding?: boolean;
};

const TopPageContainer: FC<PropsWithChildren<Props>> = ({
  children,
  padding = true,
}) => {
  return (
    <section
      className={`flex w-full flex-col gap-7 pt-16 first-of-type:pt-0 tablet:w-[calc(var(--vw)-401px)] tablet:px-0 tablet:pt-30 laptop:w-[43.0625rem] desktop:w-[64.9375rem] ${padding ? "px-4" : ""}`}
      data-testid="top-page-container"
    >
      {children}
    </section>
  );
};

export default TopPageContainer;
