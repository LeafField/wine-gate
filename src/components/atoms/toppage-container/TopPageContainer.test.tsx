import { render, screen } from "@testing-library/react";
import TopPageContainer from "./TopPageContainer";

describe("TopPageContainerの単体テスト", () => {
  test("スナップショットテスト", () => {
    const { container } = render(<TopPageContainer />);
    expect(container).toMatchSnapshot();
  });

  test("paddingがtrueの場合", () => {
    render(<TopPageContainer />);
    expect(screen.getByTestId("top-page-container")).toHaveClass("px-4");
  });

  test("paddingがfalseの場合", () => {
    render(<TopPageContainer padding={false} />);
    expect(screen.getByTestId("top-page-container")).not.toHaveClass("px-4");
  });
});
