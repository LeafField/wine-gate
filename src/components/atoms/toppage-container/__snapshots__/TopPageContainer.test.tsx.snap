// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPageContainerの単体テスト スナップショットテスト 1`] = `
<div>
  <section
    class="flex w-full flex-col gap-7 pt-16 first-of-type:pt-0 tablet:w-[calc(var(--vw)-401px)] tablet:px-0 tablet:pt-30 laptop:w-[43.0625rem] desktop:w-[64.9375rem] px-4"
    data-testid="top-page-container"
  />
</div>
`;
